handlers/oauth/oauth_test.go

```diff
@@ -0,0 +1,286 @@
+	}
+
+	var responseJSON model.UserIdentifier
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected UserIdentifier format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+}
+
+func Test_OauthCallbackReturnsExistingUser(t *testing.T) {
+	store := initStore()
+	expectedUser, createUserErr := store.CreateUser(model.User{
+		Email:    "oauthcallbackereturn@example.com",
+		Username: "iliketurtles",
+	})
+
+	if createUserErr != nil {
+		t.Fatalf("issue creating mock user %s", createUserErr)
+	}
+
+	completeUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
+		return goth.User{
+			UserID: strconv.Itoa(int(expectedUser.ID)),
+		}, nil
+	}
+	rec := httptest.NewRecorder()
+
+	ctx, _ := gin.CreateTestContext(rec)
+
+	newQuery := url.Values{
+		"provider": []string{"twitter"},
+	}
+	req, httpRequestErr := http.NewRequest("POST", "/", nil)
+	if httpRequestErr != nil {
+		t.Fatal(httpRequestErr)
+	}
+	req.URL.RawQuery = newQuery.Encode()
+	ctx.Request = req
+	OauthCallback(ctx)
+	result := rec.Result()
+	if result.StatusCode != http.StatusOK {
+		t.Fatalf("Expected %d, got %d", http.StatusOK, result.StatusCode)
+	}
+
+	var responseJSON model.UserIdentifier
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected UserIdentifier format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+
+	if responseJSON.ID != expectedUser.ID {
+		t.Fatalf("Expected user %+v but got %+v", expectedUser, responseJSON)
+	}
+}
+
+func Test_OauthCallbackCreatesNewUser(t *testing.T) {
+	store := initStore()
+	completeUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
+		rollingID++
+		return goth.User{
+			UserID: strconv.Itoa(int(rollingID)),
+			Email:  "test@example.com",
+		}, nil
+	}
+	rec := httptest.NewRecorder()
+
+	ctx, _ := gin.CreateTestContext(rec)
+
+	newQuery := url.Values{
+		"provider": []string{"twitter"},
+	}
+
+	req, httpRequestErr := http.NewRequest("POST", "/", nil)
+	if httpRequestErr != nil {
+		t.Fatal(httpRequestErr)
+	}
+	req.URL.RawQuery = newQuery.Encode()
+	ctx.Request = req
+	OauthCallback(ctx)
+	result := rec.Result()
+	if result.StatusCode != http.StatusOK {
+		t.Fatalf("Expected %d, got %d", http.StatusOK, result.StatusCode)
+	}
+
+	var responseJSON model.UserIdentifier
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected UserIdentifier format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+	expectedUser := store.Users[responseJSON.ID]
+
+	if expectedUser == nil {
+		t.Fatalf("Expected a user to be created but was nil response was %+v", responseJSON)
+	}
+}
+
+func Test_InitializeProviders(t *testing.T) {
+	setEnvErr := os.Setenv("ENABLED_PROVIDERS", "twitter")
```

Comment:
nit: Similar to above. I would name these errors as you have here in a scenario where they are all evaluated at the same time. Since these errors are all being handled sequentially after being received, I'd prefer to stick to the convention of naming it `err`.

Reply:
Done with https://github.com/leaderboardsgg/leaderboard-backend/pull/69/commits/672925b928f6e568330b0f0dba62f5406e071d46